orttraining/orttraining/python/training/torchdynamo/ort_backend.py

```diff
@@ -0,0 +1,474 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from typing import Any, Callable, Dict, List, Mapping, Set, Tuple, Type
+
+import numpy as np
+import onnx
+import torch
+import torch._prims.executor
+import torch.onnx._onnx_supported_ops
+from torch._decomp import decomposition_table
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.fx.passes.fake_tensor_prop import FakeTensorProp
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.operator_support import OperatorSupport
+from torch.fx.passes.tools_common import CALLABLE_NODE_OPS
+from torch.onnx._globals import GLOBALS as ONNX_GLOBALS
+
+import onnxruntime  # type: ignore
+from onnxruntime.capi import _pybind_state as ORTC
+
+_NP_DTYPE = {
+    torch.float16: np.float16,
+    torch.float32: np.float32,
+    torch.float64: np.float64,
+    torch.uint8: np.uint8,
+    torch.int8: np.int8,
+    torch.int16: np.int16,
+    torch.int32: np.int32,
+    torch.int64: np.longlong,
+    torch.bool: np.bool_,
+}
+
+
+def _nvtx_range_push(name: str):
+    if torch.cuda.is_available():
+        torch.cuda.nvtx.range_push(name)
+
+
+def _nvtx_range_pop():
+    if torch.cuda.is_available():
+        torch.cuda.nvtx.range_pop()
+
+
+def _get_ort_device_type(device_type: str, device_index: int):
+    if device_type == "cuda":
+        return ORTC.OrtDevice.cuda()
+    if device_type == "cpu":
+        return ORTC.OrtDevice.cpu()
+    if device_type == "ort":
+        return ORTC.get_ort_device(device_index).device_type()
+    raise Exception("Unsupported device type: " + device_type)
+
+
+logger = logging.getLogger(__name__)
+# Uncomment the following line to print out development info.
+# logger.setLevel(logging.INFO)
+
+
+def _get_onnx_supported_table() -> Set[str]:
+    onnx_supported_ops: Set[str] = set()
+    for aten_op_name, schema in torch.onnx._onnx_supported_ops.all_symbolics_schemas().items():
```

Comment:
We haven't exposed one, but I think we can! @wschin any thoughts on how you would like to interact with it / information it should provide?

Reply:
It'd be great if it can tell which `torch._ops.OpOverload` is exportable or not. I am looking for
```python
supported_opoverload: Set[torch._ops.OpOverload] = torch.onnx._onnx_supported_ops.all_opoverload()
```

Reply:
Re-opened for future reference

Reply:
Thanks. I am looking at ways to produce a more accurate set of exportable ops. The current set in `torch.onnx._onnx_supported_ops.all_symbolics_schemas` is for documentation purposes and is not entirely reliable. (or should we say, very unreliable 😅 )

Reply:
Sure.